new-tutorial/5_graph_classification.py

```diff
@@ -0,0 +1,221 @@
+# return two objects for each iteration. Th second element is simply a
+# label vector representing the category of each graph in the minibatch.
+# The first element will be described below.
+# 
+
+
+######################################################################
+# What is a batched graph?
+# ------------------------
+# 
+# The first element is a batched ``DGLGraph``, which itself is also a
+# ``DGLGraph`` (so you can still treat it as a normal ``DGLGraph`` object
+# as in :doc:`here <2_dglgraph>`). It however contains the information
+# necessary for recovering the original graphs, such as the number of
+# nodes and edges of each graph element.
+# 
+
+batched_graph, labels = batch
+print('Number of nodes for each graph element in the batch:', batched_graph.batch_num_nodes())
+print('Number of edges for each graph element in the batch:', batched_graph.batch_num_edges())
+
+# Recover the original graph elements from the minibatch
+graphs = dgl.unbatch(batched_graph)
+print('The original graphs in the minibatch:')
+print(graphs)
+
+
+######################################################################
+# Define Model
+# ------------
+# 
+# This tutorial will build a two-layer `Graph Convolutional Network
+# (GCN) <http://tkipf.github.io/graph-convolutional-networks/>`__. Each of
+# its layer computes new node representations by aggregating neighbor
+# information. If you have gone through the
+# :doc:`introduction <1_introduction>`, you will need to notice two
+# differences:
+# 
+# -  Since you will be predicting a single category for the *entire graph*
+#    instead of for every node, you will need to aggregate the
+#    representations of all the nodes and potentially the edges to form a
+#    graph-level representation. Such process is more commonly referred as
+#    a *readout*. A simple choice is to average the node features of a
+#    graph with ``dgl.mean_nodes()``.
+#
+# -  The input graph to the model will be a batched graph yielded by the
+#    ``DataLoader``. The readout functions provided by DGL can handle
+#    batched graphs so that they will return one representation for each
+#    minibatch element.
+# 
+
+from dgl.nn import GraphConv
+
+class GCN(nn.Module):
+    def __init__(self, in_feats, h_feats, num_classes):
+        super(GCN, self).__init__()
+        self.conv1 = GraphConv(in_feats, h_feats)
+        self.conv2 = GraphConv(h_feats, num_classes)
+    
+    def forward(self, g, in_feat):
+        h = self.conv1(g, in_feat)
+        h = F.relu(h)
+        h = self.conv2(g, h)
+        g.ndata['h'] = h
+        return dgl.mean_nodes(g, 'h')
+
+
+
+######################################################################
+# Training Loop
+# -------------
+# 
+# Training loop simply involves iterating over the training set with the
+# ``DataLoader`` you created and computing the gradients, just as you
+# would do for image classification or language modeling.
```

Comment:
The training loop iterates over the training set with the
``DataLoader`` object and computes the gradients, just like image classification or language modeling.